### ISSUE TITLE
Improve runtime variables override in triggers

### ISSUE THREAD

**Issue by lucafaggianelli** [mage-ai/mage-ai#1760]
**Is your feature request related to a problem? Please describe.**
Case 1: As a user I would like to propagate new global runtime variables into triggers.

Case 2: As a user I would like to override only specific runtime variables into triggers.

**Describe the solution you'd like**
*** Case 1
1. I have these global variables defined in the pipeline: `var1`, `var2`
2. I create a trigger and I enable the switch for overriding variables
3. Later I add a global var named `var3` in the pipeline
4. `var3` is not present in the trigger

This case could be solved simply running the pipeline with a dict resulting from the merge of the global vars and overridden vars:
```py
runPipeline({**global_vars, **trigger_vars})
```

*** Case 2
1. I have these global variables defined in the pipeline: `var1`, `var2`
2. I create a trigger and I enable the switch for overriding variables
3. I modify the value of `var1` in the trigger, but not the one of `var2`
4. I modify the value of `var2` in the pipeline
5. the value of `var2` in the trigger is still the old one

Basically I would like to choose which variable to override in the trigger, maybe adding a Switch in the UI near each variable in the *override variable* UI in the trigger.

Solving Case2 would solve Case1 as well

**Describe alternatives you've considered**
Modify vars by hand in the triggers

**Additional context**

**Comment by tommydangerous** [comment]
Fixed
